ui/perfherder/graphs/TestDataModal.jsx

```diff
@@ -508,10 +518,63 @@ export default class TestDataModal extends React.Component {
               <Col className="p-2 col-4">
                 <InputFilter
                   disabled={relatedTests.length > 0}
+                  placeholder="filter tests e.g. linux tp5o"
                   updateFilterText={this.applyFilters}
                 />
               </Col>
             </Row>
+            {availableTags.length > 1 && (
+              <>
+                <Row className="justify-content-start">
+                  <Col className="p-2">
+                    <FormGroup>
+                      <Label for="selectMultiTags">Tags</Label>
+                      <Input
+                        className="fa"
+                        type="select"
+                        name="selectMultiTags"
+                        id="selectMultiTags"
+                        multiple
+                      >
+                        {availableTags.sort().map((tag) => (
+                          <option
+                            key={`available-tag-${tag}`}
+                            data-testid={`available-tag ${tag}`}
+                            onClick={() => this.toggleTag(tag)}
+                          >
+                            {tag}
+                          </option>
+                        ))}
+                      </Input>
+                    </FormGroup>
+                  </Col>
+                </Row>
+                <Row className="pb-2 justify-content-start">
+                  <Col className="p-2" sm="auto">
+                    {activeTags.sort().map((tag, index) => (
+                      <React.Fragment key={`active-tag-${tag}`}>
+                        <Button
+                          id={`active-tag-${index}`}
+                          data-testid={`active-tag ${tag}`}
+                          className="mr-3"
+                          outline
+                          onClick={() => this.toggleTag(tag)}
+                        >
+                          {tag}
+                          <Badge color="btn-darker-secondary">×</Badge>
+                        </Button>
+                        <UncontrolledTooltip
```

Comment:
I think you could use the `title` attribute on the badge instead of a dedicated tooltip.